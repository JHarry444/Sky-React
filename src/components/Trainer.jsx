
const Trainer = () => {
    const me = {
        name: "JH",
        age: 29,
        specialism: "software"
    }


    return (
        <div>
            <p>Name: {me.name}</p>
            <p>Age: {me.age}</p>
            <p>Specialism: {me.specialism}</p>
            {
                //<p>Date: {new Date().toString()}</p>
            }
        </div>
    )

}

export default Trainer;