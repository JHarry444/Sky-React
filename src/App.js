// import logo from './logo.svg';
import './App.css';
import Header from './components/Header';
import Trainer from './components/Trainer';
import Conditional from './components/Conditional';
import ClassBased from './components/ClassBased';
import trainers from './data/trainers.json';
import TruthyFalsy from './components/TruthyFalsy';
import Counter from './components/Counter';
import Form from './components/Form';

function App() {
  const InnerComponent = () => <p>what up</p>;
  return (
    <div className='App'>
      <Form />
      <Counter />
      <Header />
      <TruthyFalsy error={true} />
      <TruthyFalsy error={false} />
      {
        trainers.map(({ id, name, age, specialism }) => (
          <Trainer key={id} name={name} age={age} specialism={specialism} />
        )
        )
      }
      <p>This is a paragraph</p>
      <Trainer name="JH" age={29} specialism="weaponisation" />
      {Trainer({ name: "JH", age: 29, specialism: "weaponisation" })}
      <InnerComponent />
      {InnerComponent()}
      <Conditional />
      <ClassBased />
    </div>
  );
}

export default App;
