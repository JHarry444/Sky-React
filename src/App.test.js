import { render, screen } from '@testing-library/react';
import App from './App';
import { create } from 'react-test-renderer';
import Trainer from './components/Trainer';


jest.mock("./components/CreatedDate", function () {
  return function () {
    return <p>{new Date("1994-04-04").toString()}</p>;
  };
});

describe("App tests", () => {

  test('renders learn react link', () => {
    render(<App />);
    const linkElement = screen.getByText(/learn react/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('App matches snapshot', () => {
    const app = create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });

  test('Component renders with props', () => {
    const name = "JH";
    const age = 27;
    const specialism = "software";

    const testRenderer = create(<Trainer name={name} age={age} specialism={specialism} />);
    const testInstance = testRenderer.root;

    const paragraphs = testInstance.findAllByType("p");

    expect(paragraphs[0].children).toContain(`Name: ${name}`);
    expect(paragraphs[1].children).toContain(`Age: ${age}`);
    expect(paragraphs[2].children).toContain(`Specialism: ${specialism}`);
  });
})
