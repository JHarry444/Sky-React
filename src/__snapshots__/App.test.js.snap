// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App matches snapshot 1`] = `
<div
  className="App"
>
  <h1>
    LET'S LEARN REACT
  </h1>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    THERE IS AN ERROR HERE!!!1111!!
  </p>
  <div>
    <p>
      Name: JH
    </p>
    <p>
      Age: 29
    </p>
    <p>
      Specialism: Software
    </p>
    <p>
      Mon Apr 04 1994 01:00:00 GMT+0100 (British Summer Time)
    </p>
  </div>
  <div>
    <p>
      Name: Cameron
    </p>
    <p>
      Age: 30
    </p>
    <p>
      Specialism: SDET
    </p>
    <p>
      Mon Apr 04 1994 01:00:00 GMT+0100 (British Summer Time)
    </p>
  </div>
  <div>
    <p>
      Name: Andy
    </p>
    <p>
      Age: 20
    </p>
    <p>
      Specialism: Software
    </p>
    <p>
      Mon Apr 04 1994 01:00:00 GMT+0100 (British Summer Time)
    </p>
  </div>
  <p>
    This is a paragraph
  </p>
  <div>
    <p>
      Name: JH
    </p>
    <p>
      Age: 29
    </p>
    <p>
      Specialism: weaponisation
    </p>
    <p>
      Mon Apr 04 1994 01:00:00 GMT+0100 (British Summer Time)
    </p>
  </div>
  <div>
    <p>
      Name: JH
    </p>
    <p>
      Age: 29
    </p>
    <p>
      Specialism: weaponisation
    </p>
    <p>
      Mon Apr 04 1994 01:00:00 GMT+0100 (British Summer Time)
    </p>
  </div>
  <p>
    what up
  </p>
  <p>
    what up
  </p>
  <p>
    All good
  </p>
  <p>
    OLD AND SAD!
  </p>
</div>
`;
